atlasdb-perf/src/main/java/com/palantir/atlasdb/performance/PerformanceResults.java

```diff
@@ -99,21 +99,49 @@ private static BufferedWriter openFileWriter(File file) throws FileNotFoundExcep
     }
 
     private static List<Double> getData(RunResult result) {
-        return result.getBenchmarkResults().stream()
-                .flatMap(b -> getRawResults(b.getPrimaryResult().getStatistics()).stream())
-                .collect(Collectors.toList());
+        return getRawResults(result.getPrimaryResult().getStatistics());
     }
 
     private static List<Double> getRawResults(Statistics statistics) {
+
         try {
             Field field = statistics.getClass().getDeclaredField("values");
             field.setAccessible(true);
             Multiset<Double> rawResults = (Multiset<Double>) field.get(statistics);
-            return rawResults.entrySet().stream()
-                    .flatMap(e -> DoubleStream.iterate(e.getKey(), d -> d).limit(e.getValue()).boxed())
-                    .collect(Collectors.toList());
+            return downSample(rawResults);
         } catch (NoSuchFieldException | IllegalAccessException e) {
-            return Lists.newArrayList();
+            String msg = new StringBuilder().append("Could not get values from statistics !")
```

Comment:
I guess you could extend `MultisetStatistics` and add a getter, but not sure if that is actually better.